utils/uri/matcher.go

```diff
@@ -0,0 +1,75 @@
+package uri
+
+import (
+	"errors"
+	"strings"
+)
+
+var (
+	ErrInvalidURI = errors.New("invalid URI string")
+)
+
+//hs:<service_name>:<subsystem>:<version>:<category>:<resource_name>?<allowed_arguments>#<permission_metadata>
+type URI struct {
```

Comment:
Could you add a `MarshalJSON` method for the URIs that would marshal the URIs into a string (e.g.: https://github.com/unicsmcr/hs_auth/blob/master/utils/auth/common/authLevel.go#L35).

This way we'll be able to use the `URI` struct in API response structs and the router will automatically marshal them into URIs in string form when sending the response to the user. Otherwise, the URIs in responses like this: https://github.com/unicsmcr/hs_auth/blob/master/routers/api/v2/types.go#L17 will just look like empty JSON objects.

Reply:
Good point, I'll add those too